internal/provider/physical_replication_stream.go

```diff
@@ -0,0 +1,380 @@
+		} else {
+			resp.Diagnostics.AddError(
+				"Error getting PCR stream",
+				fmt.Sprintf("Could not get PCR stream: %s", formatAPIErrorMessage(err)),
+			)
+		}
+		return
+	}
+
+	loadPCRStreamToTerraformState(apiResp, &state)
+	diags = resp.State.Set(ctx, state)
+	resp.Diagnostics.Append(diags...)
+}
+
+func (r *physicalReplicationStreamResource) Update(
+	ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse,
+) {
+	if r.provider == nil || !r.provider.configured {
+		addConfigureProviderErr(&resp.Diagnostics)
+		return
+	}
+
+	var plan, state PhysicalReplicationStream
+	diags := req.Plan.Get(ctx, &plan)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+	diags = req.State.Get(ctx, &state)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	isFailoverAt := !plan.FailoverAt.IsNull()
+	isFailoverImmediately := !plan.FailoverImmediately.IsNull()
+
+	// Validate that exactly one of failover_at or failover_immediately is set
+	if isFailoverAt == isFailoverImmediately {
+		resp.Diagnostics.AddError(
+			"Invalid failover configuration",
+			"Exactly one of failover_at or failover_immediately must be set",
+		)
+		return
+	}
+	if isFailoverImmediately && !plan.FailoverImmediately.ValueBool() {
+		resp.Diagnostics.AddError(
+			"Invalid failover configuration",
+			"If failover_immediately is set, it must be set to true",
+		)
+		return
+	}
+
+	// Create update spec with FAILING_OVER status
+	failingOverStatus := client.REPLICATIONSTREAMSTATUSTYPE_FAILING_OVER
+	updateSpec := &client.UpdatePhysicalReplicationStreamSpec{
+		Status: &failingOverStatus,
+	}
+
+	// If failover_at is specified in the plan, set it in the update spec
+	// Only set failover_at if failover_immediately is not true
+	if isFailoverAt {
+		failoverTime, err := time.Parse(time.RFC3339, plan.FailoverAt.ValueString())
+		if err != nil {
+			resp.Diagnostics.AddError(
+				"Error parsing failover time",
+				fmt.Sprintf("Could not parse failover time: %s", err),
+			)
+			return
+		}
+		updateSpec.FailoverAt = &failoverTime
+	}
+
+	// Call the update API
+	traceAPICall("UpdatePhysicalReplicationStream")
+	streamObj, _, err := r.provider.service.UpdatePhysicalReplicationStream(ctx, state.ID.ValueString(), updateSpec)
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Error updating PCR stream",
+			fmt.Sprintf("Could not update PCR stream: %s", formatAPIErrorMessage(err)),
+		)
+		return
+	}
+
+	err = retry.RetryContext(
+		ctx,
+		physicalReplicationStreamCompleteTimeout,
+		waitForPhysicalReplicationStreamStatusFunc(
+			ctx, streamObj.Id, r.provider.service, streamObj, client.REPLICATIONSTREAMSTATUSTYPE_COMPLETED,
+		),
+	)
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"PCR update failed",
+			fmt.Sprintf("PCR stream failed to update: %s", formatAPIErrorMessage(err)),
+		)
+		return
+	}
+
+	state.FailoverAt = plan.FailoverAt
+	state.FailoverImmediately = plan.FailoverImmediately
```

Comment:
Is there any way that the FailoverAt value coming back from the update response could be different from what is set in the plan?  

For example, what if FailoverAt was set manually via the api to a value before or after the plan value.  etc...

Reply:
Good question. I just checked and yes, the value could actually be updated manually via a separate API call. Should we also handle this with a separate field? Something like `failover_requested_at`?

Reply:
As discussed separately, this is fine as is.